pkg/addonmanager/manager.go

```diff
@@ -47,12 +47,12 @@ type AddonManager interface {
 	Start(ctx context.Context) error
 
 	// StartWithInformers starts all registered addon agent with the given informers.
-	StartWithInformers(ctx context.Context,
-		kubeInformers kubeinformers.SharedInformerFactory,
-		workInformers workv1informers.SharedInformerFactory,
-		addonInformers addoninformers.SharedInformerFactory,
-		clusterInformers clusterv1informers.SharedInformerFactory,
-		dynamicInformers dynamicinformer.DynamicSharedInformerFactory) error
+	// StartWithInformers(ctx context.Context,
+	// 	kubeInformers kubeinformers.SharedInformerFactory,
+	// 	workInformers workv1informers.SharedInformerFactory,
+	// 	addonInformers addoninformers.SharedInformerFactory,
+	// 	clusterInformers clusterv1informers.SharedInformerFactory,
+	// 	dynamicInformers dynamicinformer.DynamicSharedInformerFactory) error
```

Comment:
@qiujian16 @skeeey 
The only concern I have, is it safe to remove this? I didn't find any place the function is called externally.
I need to remove this or change the function siginature because in the `cloudeventsAddonManager` implementation, we can't get the `workv1informers.SharedInformerFactory` from clientHolder (from sdk-go), we can only get `workv1informers.Informer` type.

Reply:
I think we use this ocm for addon Template

Reply:
I reverted the change in `AddonManager` interface, added a mock (TODO item) implementation in the `cloudeventsAddonManager ` implementation. see: https://github.com/open-cluster-management-io/addon-framework/pull/256/commits/5c1a6e7a994360ea141655c3e8eb3bf1cf9379ed